Switch to goober to half the size of the package.

cristianbote opened the pull request:
Hi @ShaunLWM,

Was looking over the list of folks who are using goober and your repo caught my eye. I'm not sure why it was under repos that are using goober 🤷  cause I couldn't find any trace of goober in here. Anyway, right now the package has `27.6kB` by using `goober` this goes down to `13.2kB` half the size.

If this is not something that you are keen into having, no problemo! I totally understand.

Let me know if there's something else you wanna discuss about `goober` or if you have questions just pop on https://goober.rocks and open up an issue. ✌️

ShaunLWM commented:
Hello, I was trying to migrate from `styled-components` to `goober`, and probably that's why it appeared that goober was one of the dependencies. I was also trying to see if goober is a drag-and-drop replacement (but sadly not). I was able to replicate the `isClick` function but was unable to merge user's prop styles into the current styles, that's why I gave up and revert back to styled-components (time constraint :( ).

Your commit helped me alot actually. I will take some time to go through the commit and the documentation one more time but it will most probably get merged anyway.
Thank you so much for your help. Really appreciate it :)

Edit: Sorry, accidentally close the commit.

cristianbote commented:
Oh totally understandable @ShaunLWM! And also I am really not trying to push goober on you 😄  at all! If my PR feels just a tiny bit like that, I would close it instead, and I can totally understand.

To give you a summary of what I changed, basically I switched from tagged templates to array of object styles to be able to accept the user defined `styles` coming from prop:
```jsx
const Item = styled('div')(({ styles }) => [
  {
    margin: 0,
  },
  styles // styles coming from the props
]);
```
You can read more about it here https://goober.js.org/api/styled#arrays

Again, if this feels just a tiny bit pushy and something that you do not want to be involved with it's completely understandable. If there's anything else I can help out with I would gladly do it.

ShaunLWM commented:
Hey, no worries about that!  You are in no way pushy about goober. In fact, you're really helpful! Hahaha I was also trying to move over from `styled-components` to `goober` due to the library size anyway.
Thank you so much nonetheless. I will take some time out to read through goober's documentation properly (and I will accept the commit as soon as my work is cleared some time this week!)

Have a nice week ahead! :)
